ng/src/web/pages/credentials/dialog.js

```diff
@@ -114,206 +118,225 @@ class CredentialsDialog extends React.Component {
-              title={_('Replace existing privacy password with')}
-              onChange={onValueChange}/>
-          }
-          <PasswordField
-            name="privacy_password"
-            autoComplete="new-password"
-            value={privacy_password}
-            onChange={onValueChange}/>
-        </FormGroup>
-
-        <FormGroup
-          title={_('Certificate')}
-          condition={base === CLIENT_CERTIFICATE_CREDENTIAL_TYPE}>
-          <FileField
-            name="certificate"
-            onChange={onValueChange}/>
-        </FormGroup>
-
-        <FormGroup
-          title={_('Private Key')}
-          condition={
-            base === USERNAME_SSH_KEY_CREDENTIAL_TYPE ||
-            base === CLIENT_CERTIFICATE_CREDENTIAL_TYPE
-          }>
-          <FileField
-            name="private_key"
-            onChange={onValueChange}/>
-        </FormGroup>
-
-        <FormGroup
-          title={_('Auth Algorithm')}
-          condition={base === SNMP_CREDENTIAL_TYPE}>
-          <Radio
-            value={SNMP_AUTH_ALGORITHM_MD5}
-            title="MD5"
-            checked={auth_algorithm === SNMP_AUTH_ALGORITHM_MD5}
-            name="auth_algorithm"
-            onChange={onValueChange}/>
-          <Radio
-            value={SNMP_AUTH_ALGORITHM_SHA1}
-            title="SHA1"
-            checked={auth_algorithm === SNMP_AUTH_ALGORITHM_SHA1}
-            name="auth_algorithm"
-            onChange={onValueChange}/>
-        </FormGroup>
-
-        <FormGroup
-          title={_('Privacy Algorithm')}
-          condition={base === SNMP_CREDENTIAL_TYPE}>
-          <Radio
-            value={SNMP_PRIVACY_ALGORITHM_AES}
-            title="AES"
-            checked={privacy_algorithm === SNMP_PRIVACY_ALGORITHM_AES}
-            name="privacy_algorithm"
-            onChange={onValueChange}/>
-          <Radio
-            value={SNMP_PRIVACY_ALGORITHM_DES}
-            title="DES"
-            checked={privacy_algorithm === SNMP_PRIVACY_ALGORITHM_DES}
-            name="privacy_algorithm"
-            onChange={onValueChange}/>
-          <Radio
-            value={SNMP_PRIVACY_ALOGRITHM_NONE}
-            title={_('None')}
-            checked={privacy_algorithm === SNMP_PRIVACY_ALOGRITHM_NONE}
-            name="privacy_algorithm"
-            onChange={onValueChange}/>
-        </FormGroup>
-      </Layout>
+      <SaveDialog
+        visible={visible}
+        title={title}
+        onClose={onClose}
+        onSave={onSave}
+        initialData={{...DEFAULTS, ...credential}}
+      >
+        {({
+          data: state,
+          onValueChange,
+        }) => {
+          return (
+            <Layout flex="column">
+
+              <FormGroup title={_('Name')}>
+                <TextField
+                  name="name"
+                  grow="1"
+                  value={state.name}
+                  size="30"
+                  onChange={onValueChange}
+                  maxLength="80"/>
```

Comment:
Just a nitpick would be better to put the` />`  on the next (and empty) line.

Reply:
Done